tests/e2e/scale/test_scale_small_file_workload.py

```diff
@@ -1,127 +1,168 @@
 
-    return scale_leaks
+        """
+        log.info(f"The pools usage {msg} is :")
+        for entry in data:
+            log.info(f"{entry} now uses {data[entry]:,} bytes")
 
+    @pytest.mark.parametrize(
+        argnames=["file_size", "files", "threads", "interface"],
+        argvalues=[
+            pytest.param(*[8, 125000, 4, constants.CEPHFILESYSTEM]),
+            pytest.param(*[16, 125000, 4, constants.CEPHFILESYSTEM]),
+            pytest.param(*[16, 375000, 4, constants.CEPHFILESYSTEM]),
+            pytest.param(*[16, 625000, 4, constants.CEPHFILESYSTEM]),
+            pytest.param(*[16, 875000, 4, constants.CEPHFILESYSTEM]),
+            pytest.param(*[16, 1250000, 4, constants.CEPHFILESYSTEM]),
+            pytest.param(*[32, 875000, 4, constants.CEPHFILESYSTEM]),
+            pytest.param(*[32, 1250000, 4, constants.CEPHFILESYSTEM]),
+        ],
+    )
+    def test_scale_smallfile_workload(self, file_size, files, threads, interface):
+        # updating the benchmark parameters
+        self.small_files.setup_storageclass(interface)
+        self.small_files.setup_test_params(file_size, files, threads, 1)
+        self.small_files.setup_vol_size(file_size, files, threads, self.ceph_capacity)
 
-@pytest.fixture(scope="function")
-def ripsaw(request, storageclass_factory):
-    def teardown():
-        ripsaw.cleanup()
-        time.sleep(10)
+        # Verify we have enough storage capacity to run the test.
+        self.small_files.setup_operations("create")
 
-    request.addfinalizer(teardown)
+        # Run the benchmark to create files on the volume
+        self.small_files.run()
 
-    ripsaw = RipSaw()
+        # Collect pools usage after creation is done.
+        self.run_data = self.get_cephfs_data()
 
-    return ripsaw
+        # Delete the benchmark data
+        self.small_files.delete()
 
+        # Getting the usage capacity immediately after deletion
+        self.now_data = self.get_cephfs_data()
 
-@scale
-class TestSmallFileWorkloadScale(E2ETest):
-    """
-    Deploy Ripsaw operator and run different scaletests.
-    Call common smallfile_worload routie to run SmallFile workload
-    """
+        # Quarry the storage usage every 1 Min. if no difference between two
+        # samples, the backend cleanup is done.
+        still_going_down = True
+        while still_going_down:
+            log.info("Waiting for Ceph to finish cleaning up")
+            time.sleep(60)
+            self.new_data = self.get_cephfs_data()
+            still_going_down = False
+            for entry in self.new_data:
+                if self.new_data[entry] < self.now_data[entry]:
+                    still_going_down = True
+                    self.now_data[entry] = self.new_data[entry]
+
+        self.display_ceph_usage("Before ths test", self.orig_data)
+        self.display_ceph_usage("After data creation", self.run_data)
+
+        # Make sure that the test actually wrote data to the volume
```

Comment:
Metadata is stored elsewhere and there is a possibility for a leak in metadata as well. We should check if metadata is cleaned up too.

Reply:
this check is only to make sure that data was actually written to the volumes.
the verification in the end of the test that all data was deleted include the metadata as well.